client/client.go

```diff
@@ -112,6 +113,39 @@ func (client *Client) doSync(method, path string, query url.Values, body io.Read
 	return nil
 }
 
+type asyncResult struct {
+	Resource string `json:"resource"`
+}
+
+func (client *Client) doAsync(method, path string, query url.Values, body io.Reader) (string, error) {
+	var rsp response
+
+	if err := client.do(method, path, query, body, &rsp); err != nil {
+		return "", fmt.Errorf("failed to communicate with server: %v", err)
+	}
+	if err := rsp.err(); err != nil {
+		return "", err
+	}
+	if rsp.Type != "async" {
+		return "", fmt.Errorf("expected async response, got %q", rsp.Type)
+	}
+	if rsp.StatusCode != http.StatusAccepted {
+		return "", fmt.Errorf("operation not accepted")
```

Comment:
I'm not sure what you mean. Most HTTP errors will be caught by `err()`. In fact, right now there is no situation in which the server sends an `async` response without `StatusAccepted`.